services/alertmanager/alertmanager.go

```diff
@@ -328,6 +332,97 @@ func (svc *Service) configAndReload(ctx context.Context, b []byte) error {
 	return nil
 }
 
+func convertTLSConfig(id string, tls *models.TLSConfig) promconfig.TLSConfig {
+	hashedIDBytes := sha256.Sum256([]byte(id))
+	hashedID := hex.EncodeToString(hashedIDBytes[:])
+
+	caFile := tls.CAFile
+	if tls.CAFileContent != "" {
+		caFile = path.Join(alertmanagerCertDir, fmt.Sprintf("%s.ca", hashedID))
+	}
+	certFile := tls.CertFile
+	if tls.CertFileContent != "" {
+		certFile = path.Join(alertmanagerCertDir, fmt.Sprintf("%s.crt", hashedID))
+	}
+	keyFile := tls.KeyFile
+	if tls.KeyFileContent != "" {
+		keyFile = path.Join(alertmanagerCertDir, fmt.Sprintf("%s.key", hashedID))
+	}
+	return promconfig.TLSConfig{
+		CAFile:             caFile,
+		CertFile:           certFile,
+		KeyFile:            keyFile,
+		ServerName:         tls.ServerName,
+		InsecureSkipVerify: tls.InsecureSkipVerify,
+	}
+}
+
+// recreateTLSConfigFiles cleanups old tls config files and creates new ones using content from the DB.
+func (svc *Service) recreateTLSConfigFiles(chanMap map[string]*models.Channel) error {
+	des, err := os.ReadDir(alertmanagerCertDir)
+	if err != nil {
+		return errors.Wrap(err, "failed to list alertmanager certificates directory")
+	}
+	for _, de := range des {
+		if de.IsDir() {
+			continue
+		}
+
+		if err := os.Remove(path.Join(alertmanagerCertDir, de.Name())); err != nil {
+			return errors.WithStack(err)
+		}
+	}
+
+	fi, err := os.Stat(alertmanagerCertDir)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	for _, c := range chanMap {
+		webhookConfig := c.WebHookConfig
+		if webhookConfig == nil {
+			continue
+		}
+
+		httpConfig := webhookConfig.HTTPConfig
+		if httpConfig == nil {
+			continue
+		}
+
+		tlsConfig := httpConfig.TLSConfig
+		if tlsConfig == nil {
+			continue
+		}
+
+		convertedTLSConfig := convertTLSConfig(c.ID, tlsConfig)
+		for _, c := range []struct {
+			path    string
+			content string
+		}{
+			{
+				path:    convertedTLSConfig.CAFile,
+				content: tlsConfig.CAFileContent,
+			},
+			{
+				path:    convertedTLSConfig.CertFile,
+				content: tlsConfig.CertFileContent,
+			},
+			{
+				path:    convertedTLSConfig.KeyFile,
+				content: tlsConfig.KeyFileContent,
+			},
+		} {
+			if c.content != "" {
+				if err := ioutil.WriteFile(c.path, []byte(c.content), fi.Mode()); err != nil {
+					return errors.WithStack(err)
+				}
+			}
+		}
+	}
```

Comment:
IMHO if just invoke `WriteFile` three times with different params it will be easier to read.

Reply:
it is very easy to make an error when you copy-paste things, so I used this approach. But I could rework, let's ask @BupycHuk for his opinion 🤔

Reply:
a bit simplified with using a map, please take a look.

Reply:
Looks ok to me 👌